lib/database/sthDatabase.js

```diff
@@ -313,6 +313,60 @@ function getCollection(params, options, callback) {
         return process.nextTick(callback.bind(null, error));
     }
 
+    //Call fetchCollection after creating database connection if connection was not restored after database restarted
+    if (isConnectionAlive()) {
+        fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+    } else {
+        reconnectDB(callback, function() {
+            fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+        });
+    }
+}
+
+/**
+ * Check whether database is connected or not
+ */
+function isConnectionAlive() {
+    let dbConnected = true;
+    if (db == null || !db.serverConfig.isConnected()) {
```

Comment:
Better check `db === null` ?

Reply:
@AlvaroVega, I have check the same in commit https://github.com/telefonicaid/fiware-sth-comet/pull/574/commits/ba0d4e6b80232ceaaa5bc7e2a5bb9247b76602b1.